tests/idris2/linear014/Issue55Extra.idr

```diff
@@ -0,0 +1,9 @@
+1
+foo : (1 b : Bool) -> Bool
+foo b = b
+
+bar : Bool
+bar = q
+  where
+    q : Bool
+    q = foo True
```

Comment:
Hey @gallais, I added an extra test case as to why I thought it was best to propagate the RigCount in InExpr, in particular the usage of a linear term in a non-linear context.